### DISCUSSION TITLE
[Question] Client Hints

### DISCUSSION THREAD

**Discussion by mimmi20** [andreruffert/progressive-image-element#72] [Q&A] [answered]
It is possible to enable Client Hints in Chrome, to get information about the current connection. Does this have any effect on the savedata option?

**Comment by andreruffert** [comment, marked as answer]
> It is possible to enable Client Hints in Chrome, to get information about the current connection. Does this have any effect on the savedata option?

Short answer is no.

The current implementation is using the [`navigator.connection`](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/connection) API with a string comparison of the [effectiveType](https://developer.mozilla.org/en-US/docs/Web/API/NetworkInformation/effectiveType).
The following values `slow-2g|2g|3g` enable the savedata boolean by default. [More info](https://github.com/andreruffert/progressive-image-element/blob/master/progressive-image-element.js#L31-L34).

**Reply by andreruffert** [reply]
@mimmi20 if this answered your question I'd appreciate if you could mark the discussion as answered. Thanks! ✌️
